### PR TITLE
JPERF-783: Migrate CI to GitHub Actions

### PR DESCRIPTION
**Pull request by mzyromski-atlassian**
Currently used CI system (Travis CI) stopped working and we're trying to migrate to GHA anyway for all the modules on GH anyway. The configuration used here is identical as used in many other modules, e.g. https://github.com/atlassian/jvm-tasks/blob/master/.github/workflows/ci.yml.

### CODE REVIEW

**Comment by mzyromski-atlassian** [comment]
A proof that it works on my fork: https://github.com/mzyromski-atlassian/concurrency/runs/5394804530?check_suite_focus=true. Once I get approval(s) I'll remove Travis CI as a check, merge it and add GHA as a new check.

**Review by szymonra — APPROVED** [review]

**Pull request merged**
